### PR TITLE
Update Agroal to 2.5

### PR DESCRIPTION
**Pull request by graben**
@geoand: Long awaiting first update. More to come ...

### CODE REVIEW

File: narayana-spring-boot-starter-it/src/test/java/dev/snowdrop/boot/narayana/generic/GenericRecoveryIT.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ void testCrashBeforeCommit() throws Exception {
         assertEntriesAfterCrash(this.entriesService.getEntries());
 
         await("Wait for the recovery to happen")
-                .atMost(Duration.ofSeconds(60))
+                .atMost(Duration.ofSeconds(30))
```
Comment (geoand) [inline narayana-spring-boot-starter-it/src/test/java/dev/snowdrop/boot/narayana/generic/GenericRecoveryIT.java:120]: Why this change?

Comment (graben) [reply narayana-spring-boot-starter-it/src/test/java/dev/snowdrop/boot/narayana/generic/GenericRecoveryIT.java:120]: Changed back to 30 because a better workaround is now available with Agroal 2.5. look at flushonclose. It's only for a stupid behavior of H2 with shared connections between recovery and normal purposes.

**Review by geoand — APPROVED** [review]
